test/resources/Invoices.spec.js

```diff
@@ -137,18 +137,11 @@ describe('Invoices Resource', function() {
           url: '/v1/invoices/upcoming?customer=cus_123&subscription=sub_123',
           headers: {},
           data: {
-            subscription_items: {
-              0: {
-                plan: 'potato',
-              },
-              1: {
-                plan: 'rutabaga',
-              },
-              2: {
-                deleted: true,
-                id: 'SOME_ID',
-              },
-            },
+            subscription_items: [
+              {plan: 'potato'},
+              {plan: 'rutabaga'},
+              {id: 'SOME_ID', deleted: true},
+            ],
```

Comment:
I'm somewhat worried about our test coverage reduction here... is there anything in our test suite to verify that these POST bodies are form-encoded with `subscription_items[0][plan]` instead of `subscription_items[][plan]` ?

Reply:
Fair! Looking into it.

Reply:
Okay, so this is difficult to test in the current state. The URL-encoding of the request parameters happens in the original `_request` method, but `getSpyableStripe()` returns an instance of `stripe` where `_request` is patched to memorize the method parameters instead of actually sending the request.

We've recently added `nock` for testing the telemetry stuff, so we could probably write some new tests that use it to check the actual body of our HTTP requests.

Reply:
Yeah, I think nock sounds like the right tool for this job, should be pretty straightforward - fft DM me or paul if you need help with it!

Reply:
So I lied, we didn't add nock at all. No idea why I thought that was the case!

Nevertheless, I tried writing a test with nock to test `StripeResource._request`, but I can't seem to have nock intercept the request -- for some reason the request is always sent to the real API server. Mind giving it a try @rattrayalex-stripe?

Reply:
We add nock in https://github.com/stripe/stripe-node/pull/559 😄